### PR TITLE
JS Custom Messaging

### PR DESCRIPTION
**Pull request by howhelan**

### CODE REVIEW

**Comment by capt-haddock** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **howhelan**
You can assign the PR to them by writing `/assign @howhelan` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=pauwels-labs%2Fredact-client).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/pauwels-labs/redact-client/blob/mutual-tls2/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by capt-haddock** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **howhelan**
You can assign the PR to them by writing `/assign @howhelan` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=pauwels-labs%2Fredact-client).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/pauwels-labs/redact-client/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: src/routes/data/get.rs
Comment (ajpauwels) [inline src/routes/data/get.rs:84]: no unwrap

File: src/routes/data/get.rs
Comment (ajpauwels) [inline src/routes/data/get.rs:79]: Instead of declaring sanitized_message here and making it mut, there's a more functional way of doing this using and_then:
```
let sanitized_message = query_params.js_message.and_then(|m| {
    let decoded_param = ...
});
```

File: src/routes/data/get.rs
Comment (ajpauwels) [inline src/routes/data/get.rs:218]: Sanitization should happen on this route as well, even if it's been done in the without_tokens route and should be sanitized already

File: src/routes/data/post.rs
Comment (ajpauwels) [inline src/routes/data/post.rs:147]: Shouldn't post also have a message passed?

File: static/secure.handlebars
Comment (ajpauwels) [inline static/secure.handlebars:28]: Can we merge these two?

**Review by ajpauwels — CHANGES_REQUESTED** [review]

File: src/routes/data/get.rs
Comment (howhelan) [reply src/routes/data/get.rs]: done

File: src/routes/data/get.rs
Comment (howhelan) [reply src/routes/data/get.rs:84]: remvoed

File: src/routes/data/post.rs
Comment (howhelan) [reply src/routes/data/post.rs:147]: yep, fixed

**Comment by capt-haddock** [comment]
@howhelan: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-test | 88d9ad9189e07f53a92f30929f128a9676627a14 | [link](http://dashboard.dev.pauwelslabs.com/namespaces/jx/pipelineruns/bs-redact-client-pr-34-unit-test-btlgd) | `/test unit-test`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by capt-haddock** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[ajpauwels](https://github.com/pauwels-labs/redact-client/pull/34#issuecomment-887632781 "Approved")*

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=pauwels-labs%2Fredact-client).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/pauwels-labs/redact-client/blob/main/OWNERS)~~ [ajpauwels]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
